packages/astro/components/ViewTransitions.astro

```diff
@@ -50,6 +50,14 @@ const { fallback = 'animate' } = Astro.props;
 		return el.dataset.astroReload !== undefined;
 	}
 
+	function isCrossOrigin(action?: string): boolean {
+		if (!action) {
+			return false;
+		}
+		const actionURL = new URL(action, location.href);
+		return actionURL.hostname !== location.hostname;
```

Comment:
cross origin is more than the host. It should also include protocol and port.
Better use actionURL.origin !== location.origin?